integration-tests/connect_test.go

```diff
@@ -45,6 +46,40 @@ func TestGoCqlConnect(t *testing.T) {
 	require.Equal(t, "fake", iter.Columns()[0].Name)
 }
 
+func TestProtocolVersionNegotiation(t *testing.T) {
+	testCassandraVersion := env.CassandraVersion
+	env.CassandraVersion = "2.1" // downgrade C* version for protocol negotiation test
```

Comment:
we shouldn't do this, we should just add a function that creates a simulacron cluster with the provided version instead of using the env var. Maybe this should even have multiple test cases where you provide the inputs: `C* version, expected protocol version in control connection`, and you can create a function in the control connection to return the current protocol version to assert on this test

Reply:
Also I'm not sure if simulacron will respect the C* version when considering which protocol versions to use. We might have to use the "fake" test server (don't really need CCM here, the "fake" test server should work just fine and would make the tests much faster).